### ISSUE TITLE
Heapster not compatible with kubeclient that uses v1beta3 

### ISSUE THREAD

**Issue by vishh** [GoogleCloudPlatform/heapster#233]
I see List failures when heapster uses v1beta3 client library. I haven't looked at the client library differences when it comes to kubernetes API versions. This behavior is odd since the kube client exposes internal API objects.

```shell
2015-04-13T21:58:28.706955553Z + EXTRA_ARGS=
2015-04-13T21:58:28.706992235Z + '[' '!' -z --kubernetes_version=v1beta3 ']'
2015-04-13T21:58:28.706992235Z + EXTRA_ARGS=--kubernetes_version=v1beta3
2015-04-13T21:58:28.706992235Z + '[' '!' -z 10.0.0.1 ']'
2015-04-13T21:58:28.706992235Z + EXTRA_ARGS='--kubernetes_master 10.0.0.1:80 --kubernetes_version=v1beta3'
2015-04-13T21:58:28.706992235Z + HEAPSTER=/usr/bin/heapster
2015-04-13T21:58:28.706992235Z + case $SINK in
2015-04-13T21:58:28.706992235Z + HEAPSTER='/usr/bin/heapster --sink influxdb'
2015-04-13T21:58:28.706992235Z + '[' '!' -z 10.0.0.1 ']'
2015-04-13T21:58:28.706992235Z + INFLUXDB_ADDRESS=
2015-04-13T21:58:28.706992235Z + '[' '!' -z 10.0.238.82 ']'
2015-04-13T21:58:28.706992235Z + INFLUXDB_ADDRESS=10.0.238.82:80
2015-04-13T21:58:28.706992235Z + /usr/bin/heapster --sink influxdb --sink_influxdb_host 10.0.238.82:80 --kubernetes_master 10.0.0.1:80 --kubernetes_version=v1beta3
2015-04-13T21:58:28.721064549Z I0413 21:58:28.721008       7 heapster.go:45] /usr/bin/heapster --sink influxdb --sink_influxdb_host 10.0.238.82:80 --kubernetes_master 10.0.0.1:80 --kubernetes_version=v1beta3
2015-04-13T21:58:28.721130318Z I0413 21:58:28.721120       7 heapster.go:46] Heapster version 0.10.0
2015-04-13T21:58:28.721592978Z I0413 21:58:28.721568       7 kube.go:237] Using Kubernetes client with master "http://10.0.0.1:80" and version v1beta3
2015-04-13T21:58:28.721644965Z I0413 21:58:28.721636       7 kube.go:238] Using kubelet port "10250"
2015-04-13T21:58:28.721684541Z I0413 21:58:28.721676       7 driver.go:164] Using influxdb on host "10.0.238.82:80" with database "k8s"
2015-04-13T21:58:28.725709614Z I0413 21:58:28.725696       7 heapster.go:57] Starting heapster on port 8082
2015-04-13T21:58:28.737456320Z E0413 21:58:28.737438       7 reflector.go:115] Failed to list *api.Node: minions "" not found
2015-04-13T21:58:29.739191916Z E0413 21:58:29.739112       7 reflector.go:115] Failed to list *api.Node: minions "" not found
2015-04-13T21:58:30.741866796Z E0413 21:58:30.741815       7 reflector.go:115] Failed to list *api.Node: minions "" not found
2015-04-13T21:58:31.743400948Z E0413 21:58:31.743330       7 reflector.go:115] Failed to list *api.Node: minions "" not found
2015-04-13T21:58:32.744916601Z E0413 21:58:32.744860       7 reflector.go:115] Failed to list *api.Node: minions "" not found
2015-04-13T21:58:33.746331408Z E0413 21:58:33.746234       7 reflector.go:115] Failed to list *api.Node: minions "" not found
2015-04-13T21:58:34.747989649Z E0413 21:58:34.747935       7 reflector.go:115] Failed to list *api.Node: minions "" not found
2015-04-13T21:58:35.749541932Z E0413 21:58:35.749479       7 reflector.go:115] Failed to list *api.Node: minions "" not found
2015-04-13T21:58:36.751084129Z E0413 21:58:36.751028       7 reflector.go:115] Failed to list *api.Node: minions "" not found
```

**Comment by biswars** [comment]
I observe the same issue with v1beta3. However, I am trying to get heapster run on kubernetes. Can you let me know which versions of the kube clients can I go for along with heapster ? So that I can set it up without exploring with various versions. Can I go for kube client 0.15 and master branch of heapster ?

**Comment by vishh** [comment]
Kindly use `vebeta1` as the runtime version for heapster. That is the default as of now.

**Comment by vishh** [comment]
This has been fixed in v0.13.0 release
